### ISSUE TITLE
Connection state/daemon health

### ISSUE THREAD

**Issue by kopierschnitte** [zivillian/ism7mqtt#142]
Hello,
thanks for the great work. I have a feature request regarding the overall status of the ism7mqtt process:
It would be great if it could also report some general states about itself. I'm thinking about

- connection state between process and ISM / is it alive?
- if the process is still running or has been crashed (using MQTT last will?)
- stdout/stderr output to MQTT

As I'm running the process headless on a RPi, this would make it easier to monitor the service using ioBroker and become alerted if the values aren't recent anymore, for example.

**Comment by krusta4711** [comment]
In case the feature request will not be implemented here are two tips how you can help yourself:

1. Let run ism7mqtt either as systemctl service or in a Docker container. Both variations can be configured to automatically restart ism7mqtt in case it stops working
2. For the alive connection there are two scenarios which might happen:
2.1 The connection to the ism module gets lost because the module is closing it: in that case the  ism7mqtt simply stop working. Solution: see point 1.
2.2 The connection is alive but the is module does not send any data over it anymore (see #115 ). For this case you could write an own observer in ioBroker to restart the ism module.

**Comment by kopierschnitte** [comment]
Thanks. As I already use systemctl, I'm quite sure, 2.1 can already be ruled out. I didn't know that the process terminates itself in this case ;-)
So there's only the 2nd issue left. I'll see if I can develop a script for this case.

**Comment by krusta4711** [comment]
How to reboot the ism module by a curl command I described here:
https://github.com/zivillian/ism7mqtt/issues/115#issuecomment-2350964304

**Comment by kopierschnitte** [comment]
Perfect, thanks. Now I need to find a reliable indicator for such a situation ;-)

**Comment by krusta4711** [comment]
I started with some often changing temperature values but changed it recently to use the time/clock (parameter 220032/Uhrzeit). That is sent every minute. Sometimes a minute is missing, so I wait for 5 minutes before I reboot the ism module.

**Comment by kopierschnitte** [comment]
Good point. I would still like to leave this request open. Maybe, it could be handled within the application's watchdog logic ;-)

**Comment by Eba-M** [comment]
I run ism7mqtt within screen using a script. if ism7mqtt stops, it will restart at once.
